### DISCUSSION TITLE
What is unit testing?

### DISCUSSION THREAD

**Discussion by PPartisan** [PPartisan/Roky#23] [Q&A]
Many parts of this project mention the importance of unit testing. What is unit testing, and why is it important? The project uses `JUnit5` and `Kotest` - what are these, and how are they relevant to unit testing? Where can I learn more about writing unit tests with these frameworks? What makes a "good" unit test?

**Comment by TotallyPresent** [comment]
In looking for resources I found a good series of debates on unit testing here and I wondered what you think? I will post.

**Reply by TotallyPresent** [reply]
[![Why I Don’t Unit Test](https://img.youtube.com/vi/ZGKGb109-I4/0.jpg)](https://youtu.be/ZGKGb109-I4?si=a4RFr9HQnlB7Tofq "Why I Don’t Unit Test")

**Reply by TotallyPresent** [reply]
[![We Finally Agree On Unit Tests](https://img.youtube.com/vi/MbU-PKukdMw/0.jpg)](https://www.youtube.com/watch?v=MbU-PKukdMw "Why I Don’t Unit Test")

**Reply by TotallyPresent** [reply]
[![YOU DONT UNIT TEST??? DevHour #1 Theo](https://img.youtube.com/vi/pvBHyip4peo/0.jpg)](https://www.youtube.com/watch?v=pvBHyip4peo "YOU DONT UNIT TEST??? DevHour #1 Theo")
